### PR TITLE
Simplifies Secrets Script

### PR DESCRIPTION
**Pull request by redreceipt**
This makes the bash script actually good code :)

**IMPORTANT: If this is for testing code in our npm dependencies, you should be pointing your PR at the `canary` or `next` branch. `master` should never rely on unreleased code.**

### CODE REVIEW

File: scripts/secrets.sh [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+
+SECRETS=(
+	"apolloschurchapp/.env.shared"
+	"apollos-church-api/.env.shared"
+	"apolloschurchapp/android/key.json"
+	"apolloschurchapp/android/app/apollos.keystore"
+)
+
+for file in "${SECRETS[@]}"; do
+	if [ "$1" = "-e" ]; then
```
Comment (christinamegura) [inline scripts/secrets.sh:23]: Where would this "-e" come from?

**Comment by christinamegura** [comment]
I'm new to looking at bash scripts but it is very easy to read and follow. How would I go about testing this?

**Comment by redreceipt** [comment]
Awesome! Yea that -e is the first command line argument passed to the script. You would test this with yarn unpack-secrets <KEY> and then investigate the four files that were decrypted. The key comes from a file I put in Apollos Support project in Basecamp.

**Comment by christinamegura** [comment]
`yarn unpack-secrets -e` encrypts all the files 👍 

`yarn unpack-secrets <KEY>` decrypts mostly everything except for about half of the apollos.keystore file. I don't know what to make of that

**Comment by redreceipt** [comment]
Well technically you encrypt with yarn pack-secrets but maybe that’s confusing and could be better?

**Comment by christinamegura** [comment]
Ope, duh! How do you feel about `yarn secrets`(...-e; ...<KEY>) instead? `pack` with `-e` might be a little redundant? 

Should the entire apollos.keystore file decrypt? Only a section of it did when I ran the command.

**Comment by redreceipt** [comment]
you don't have to specify the -e. it's just 

`yarn pack-secrets <KEY>`
`yarn unpack-secrets <KEY>`

yea I'm not suprised a portion of the keystore is further encrypted, that's its purpose on Google's end

**Comment by redreceipt** [comment]
maybe this clears it up, to answer your original question, that -e is built into the `yarn pack` command, see the package.json

**Comment by redreceipt** [comment]
I made that change to the API, I think you're right, I like that better.

**Comment by christinamegura** [comment]
Okay! 5th time is the charm.

`yarn unpack-secrets <KEY>` decrypts the four files, with apollos.keystore having a partial decrypt ✅

`yarn pack-secrets <KEY>` creates four new (edited) encrypted files ✅

This is clear and easy to follow. I'm chalking the confusion up to a personal problem since this is my first swing at a bash script and I didn't check everywhere I needed to at first.

**Comment by christinamegura** [comment]
I like the API change but now when I open the four files they are already decrypted (not sure why this is happening now?). `yarn secrets -e <KEY>` will encrypt the files but then saves them as encoded (ie env.shared.enc). I don't think this was the case before and I cannot confirm the decrypt is working.


If you revert back to `pack-secrets` and `unpack-secrets` I have confirmed that working as follows:
- `yarn unpack-secrets <KEY>` decrypts the four files, with apollos.keystore having a partial decrypt 
- `yarn pack-secrets <KEY>` creates four new (edited) enctyped files

**Comment by redreceipt** [comment]
What you described is how it’s supposed to work

**Review by christinamegura — APPROVED** [review]

**Pull request merged**